src/tango/quic/fd_quic.c

```diff
@@ -327,13 +320,34 @@ fd_quic_init( fd_quic_t * quic ) {
   fd_quic_limits_t const * limits = &quic->limits;
   fd_quic_config_t       * config = &quic->config;
 
-  if( FD_UNLIKELY( !config->role         ) ) { FD_LOG_WARNING(( "cfg.role not set"      )); return NULL; }
-  if( FD_UNLIKELY( !config->idle_timeout ) ) { FD_LOG_WARNING(( "zero cfg.idle_timeout" )); return NULL; }
+  if( FD_UNLIKELY( !config->role          ) ) { FD_LOG_WARNING(( "cfg.role not set"      )); return NULL; }
+  if( FD_UNLIKELY( !config->idle_timeout  ) ) { FD_LOG_WARNING(( "zero cfg.idle_timeout" )); return NULL; }
+
+  if( FD_UNLIKELY( (!quic->cert_object) | (!quic->cert_key_object) ) ) {
+    /* FIXME remove this hack by separating TLS and QUIC management.
+             User should provide pre-initialized SSL_CTX with cert
+             installed.  fd_quic should not touch any certificate
+             handling. */
+    FD_LOG_WARNING(( "Warning: Certificate or key not set. Generating random." ));
+    if( FD_UNLIKELY( quic->cert_object     ) ) X509_free    ( (X509 *)quic->cert_object         );
+    if( FD_UNLIKELY( quic->cert_key_object ) ) EVP_PKEY_free( (EVP_PKEY *)quic->cert_key_object );
+
+    /* Generate certificate key */
+    uchar cert_private_key[ 32 ];
+    FD_TEST( 1==RAND_bytes( cert_private_key, 32 ) );
```

Comment:
oh i should do this in my own code lol for some reason i only associated FD_TEST with tests